xdsl/dialects/bufferization.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any
+
+from xdsl.dialects.builtin import (
+    ContainerType,
+    IndexType,
+    MemRefType,
+    ShapedType,
+    TensorType,
+    UnitAttr,
+    UnrankedMemrefType,
+    UnrankedTensorType,
+)
+from xdsl.ir import Attribute, Dialect, Operation, SSAValue
+from xdsl.irdl import (
+    AnyOf,
+    AttrSizedOperandSegments,
+    IRDLOperation,
+    irdl_op_definition,
+    operand_def,
+    opt_operand_def,
+    opt_prop_def,
+    result_def,
+    var_operand_def,
+)
+
+
+@irdl_op_definition
+class AllocTensorOp(IRDLOperation):
+    name = "bufferization.alloc_tensor"
+
+    dynamic_sizes = var_operand_def(IndexType())
+    copy = opt_operand_def(AnyOf((TensorType, UnrankedTensorType)))
+    size_hint = opt_operand_def(IndexType())
+
+    tensor = result_def(AnyOf((TensorType, UnrankedTensorType)))
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True)]
+
+    def __init__(
+        self,
+        result_type: Attribute,
+        dynamic_sizes: list[Operation | SSAValue] | None = None,
+        copy: SSAValue | Operation | None = None,
+        size_hint: SSAValue | Operation | None = None,
+    ):
+        super().__init__(
+            operands=(dynamic_sizes, copy, size_hint),
+            result_types=(result_type,),
+        )
+
+    @classmethod
+    def static_type(cls, type: Attribute) -> "AllocTensorOp":
```

Comment:
What's the use of this?

Reply:
Just being explicit in user code. Felt natural to me while writing it to use it directly, cf #2301.
Do you want it removed?

Reply:
There are a few things that threw me off here. One is not using Self (from typing_extensions), and using a string instead. One is the name, although I now understand what it means. Maybe `static_shape` could be more explicit? IDK. But also the fact that the implementation is shorter than the method call, I'm not sure that the integral of confusion over time is lowered with this helper...

Reply:
Will remove